auth_matrix/admin.py

```diff
@@ -40,15 +43,22 @@ def dehydrate(self, user):
         return data
 
     def export(self, queryset, *args, **kwargs):
+        queryset = self.get_queryset()
         dataset = super().export(queryset, *args, **kwargs)
         group_names = [group.name for group in Group.objects.all()]
         for group_name in group_names:
             dataset.append_col(
-                [user.groups.filter(name=group_name).exists() for user in queryset],
+                [
+                    user.groups.filter(name=group_name).exists()
+                    for user in User.objects.all()
+                ],
                 header=group_name,
             )
```

Comment:
this is a `n+1` query - 1 query to retrieve all the groups, and then for each group a query is done to retrieve the users. The more groups you have, the worse performance gets.

Instead, you can annotate each user to get a list of its group names with each user object.

```py
group_names_subquery = (
    Group.objects
    .filter(user=OuterRef('pk'))
    .values('user')
    .annotate(group_names=ArrayAgg('name'))
    .values('group_names')
)
queryset = queryset.annotate(group_names=Subquery(group_names_subquery))
```
This results in 1 query to retrieve all the users with the group names attached to it.

Next, you can (efficiently) loop over the group names to add the extra columns (that will be query 2):

```py
for group_name in Group.objects.values_list("name", flat=True).order_by("name"):
    column = [group_name in user.group_names for user in queryset]
    ...
```
make sure to loop over `queryset` again since it will re-use the queryset cache and not create additional queries (and to guarantee the ordering, see paragraph below).

Note that you must make sure to iterate over the same `queryset` to make sure the order of rows added in the extra column is stable! If you sort the admin on `-username` for example, and `User.objects.all()` ordering is undefined (let's assume it orders on PK then, but it might be random), then your matrix will be wrong. I'd like to see a test for this!